### DISCUSSION TITLE
agent_id empty in table alerts_alert

### DISCUSSION THREAD

**Discussion by Nini81** [amidaware/tacticalrmm#616] [Q&A]
Using Tactical RMM v0.7.2
When alert is poping, the field "agent_id" is empty in the "alerts_alert" table, So we're loosing the relation between the alert and the host. There's the hostname in the message, but if 2 clients use the same hostname ...

**Comment by silversword411** [comment]
Can you provide a screenshot with the problem?

**Comment by Nini81** [comment]
[https://imgur.com/a/PWh2h0p](https://imgur.com/a/PWh2h0p)

here is it

Edit : i just realize there's the assigned_check_id that is related to the checks_check table, which contains the agent_id if needed. So my report is not really a big deal. should have seen that one earlier.

**Comment by wh1te909** [comment]
what exactly is the bug i don't understand

**Comment by Nini81** [comment]
i thought taht the "agent_id" should contain the id of the agent that poped the alert, no ?

**Comment by silversword411** [comment]
That doesn't look like a standard TRMM screen. Are you using a 3rd party tool like grafana to look at the postgres database?

**Comment by sadnub** [comment]
If the alert is triggered by a check or task, you need to get the agent through those relationships.

**Comment by wh1te909** [comment]
moving to github discussions, since this is not a bug or feature request.
there is no `agent_id` in the Alert db model so not sure where you're getting that from
`agent_id` can be accessed from the `agent` foreign key field on the alert model. so it would be something like `alert.agent.agent_id`

https://github.com/wh1te909/tacticalrmm/blob/62be0ed936c84c8b35c9951c867edd3b772b034d/api/tacticalrmm/alerts/models.py#L35

**Comment by sadnub** [comment]
@Nini81 There is an assigned_task and assigned_check relationship to the task or check that triggered the alert. Those two relationships have an agent foreign key. Do if a check triggered an alert, it would look like: alert.assigned_check.agent.hostname
